src/posix/platform/settings_file.cpp

```diff
@@ -46,16 +46,27 @@
 #include "common/debug.hpp"
 #include "posix/platform/settings_file.hpp"
 
+void platformSettingsInit(const char *aDataPath) { ot::Posix::SettingsFile::SetSettingsPath(aDataPath); }
+
 namespace ot {
 namespace Posix {
 
+char SettingsFile::sSettingsPath[] = OPENTHREAD_CONFIG_POSIX_SETTINGS_PATH;
+
+const char *SettingsFile::GetSettingsPath(void) { return sSettingsPath; }
+void        SettingsFile::SetSettingsPath(const char *aSettingsPath)
+{
+    snprintf(sSettingsPath, sizeof(sSettingsPath), "%s", aSettingsPath);
+}
+
 otError SettingsFile::Init(const char *aSettingsFileBaseName)
 {
     otError     error     = OT_ERROR_NONE;
-    const char *directory = OPENTHREAD_CONFIG_POSIX_SETTINGS_PATH;
+    const char *directory = GetSettingsPath();
 
-    OT_ASSERT((aSettingsFileBaseName != nullptr) && (strlen(aSettingsFileBaseName) < kMaxFileBaseNameSize));
-    strncpy(mSettingFileBaseName, aSettingsFileBaseName, sizeof(mSettingFileBaseName) - 1);
+    OT_ASSERT((aSettingsFileBaseName != nullptr) && ((strlen(directory) + kSlashLength + strlen(aSettingsFileBaseName) +
```

Comment:
There are `kMaxFileBasePathNameSize` and `kMaxFileBaseNameSize` const defined, here may consider two OT_ASSERT respectively

```
OT_ASSERT(strlen(directory) < kMaxFileBasePathNameSize);
OT_ASSERT((aSettingsFileBaseName != nullptr) && strlen(aSettingsFileBaseName) < kMaxFileBaseNameSize));
```